### PR TITLE
Sync upstream and downstream

### PR DESCRIPTION
**Pull request by heyselbi**
#### Motivation
https://issues.redhat.com/browse/RHODS-6136

#### Modifications
PR #7 already exists in ODH as MR #54 . Remaining commits do no clash with ODH.

#### Result
Expected result - better sync between upstream and downstream.

### CODE REVIEW

File: odh-model-controller/controllers/inferenceservice_route.go [hunk 1]
```diff
@@ -73,6 +73,11 @@ func NewInferenceServiceRoute(inferenceservice *inferenceservicev1.InferenceServ
 			Termination:                   routev1.TLSTerminationReencrypt,
 			InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyRedirect,
 		}
+	} else {
```
Comment (anishasthana) [inline odh-model-controller/controllers/inferenceservice_route.go:76]: The odh model controller code has been moved out of this repo -- we can drop this commit.

**Review by anishasthana — CHANGES_REQUESTED** [review]

File: manifests/opendatahub/rbac/networkpolicy-etcd.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: etcd
+spec:
+  podSelector:
+    matchLabels:
+      app: model-mesh
```
Comment (Jooho) [inline manifests/opendatahub/rbac/networkpolicy-etcd.yaml:21]: does it need to change to `component: model-mesh-etcd`

Comment (heyselbi) [reply manifests/opendatahub/rbac/networkpolicy-etcd.yaml:21]: @rimolive could you confirm which one is the correct one?

Comment (rimolive) [reply manifests/opendatahub/rbac/networkpolicy-etcd.yaml:21]: Should be correct, because this was tested by multiple users

Comment (heyselbi) [reply manifests/opendatahub/rbac/networkpolicy-etcd.yaml:21]: @rimolive confirmed the `app: model-mesh` label since it is for model-mesh pods

Comment (heyselbi) [reply manifests/opendatahub/rbac/networkpolicy-etcd.yaml:21]: Opened a separate issue for it: https://github.com/opendatahub-io/modelmesh-serving/issues/59

**Comment by anishasthana** [comment]
This is replaced by #60

**Comment by anishasthana** [comment]
superseded by other PRs.

**Pull request closed without merging**